File: 프로그래머스/2/42578. 의상/의상.swift
import Foundation

func solution(_ clothes:[[String]]) -> Int {
    var dict = clothes.reduce(into: [String: Int]()) { $0[$1[1], default: 0] += 1 }
    return dict.values.reduce(1) { $0 * ($1 + 1) } - 1
}

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<vector<string>> clothes) {
    unordered_map<string, int> mp;
    for(const vector<string>& c : clothes) {
        mp[c[1]]++;
    }
    int ret = 1;
    for(const auto& it: mp) {
        ret *= it.second + 1;
    }
    return ret - 1;
}